library/Icinga/Web/StyleSheet.php

```diff
@@ -149,6 +149,20 @@ protected function collect()
                 $this->lessCompiler->setTheme($this->pubPath . '/css/themes/' . $theme . '.less');
             }
         }
+
+        if ($user = Auth::getInstance()->getUser()) {
+            $mode = $user->getPreferences()->getValue('icingaweb', 'mode');
+
+            if (($theme === null || $theme === self::DEFAULT_THEME) && empty($mode)) {
```

Comment:
Why are you checking the theme here?

Reply:
because when you log in again, the default theme and dark mode are automatically selected. But `$mode = $user->getPreferences()->getValue('icingaweb', 'mode', 'none');` always returns `‘none'`. I suspect this is because dark mode is shown selected but has not yet been saved in form. Is there another way to do this?

Reply:
I don't understand this. It is of course required to save before logging out and back in. If you get `none` using the call above, it is either configured this way or saving the form didn't persist the chosen mode.